File: app/src/test/java/de/bahnhoefe/deutschlands/bahnhofsfotos/mapsforge/MarkerBitmapTest.java
package de.bahnhoefe.deutschlands.bahnhofsfotos.mapsforge;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;

import android.content.Context;

import org.junit.jupiter.api.Test;
import org.mapsforge.core.graphics.Bitmap;
import org.mapsforge.core.graphics.Paint;
import org.mapsforge.core.model.Point;

/**
 * Tests for {@link MarkerBitmap}.
 */
public class MarkerBitmapTest {

    private final Bitmap srcWithoutPhoto = mock(Bitmap.class);
    private final Bitmap srcWithPhoto = mock(Bitmap.class);
    private final Bitmap srcOwnPhoto = mock(Bitmap.class);
    private final Bitmap srcWithoutPhotoInactive = mock(Bitmap.class);
    private final Bitmap srcWithPhotoInactive = mock(Bitmap.class);
    private final Bitmap srcOwnPhotoInactive = mock(Bitmap.class);
    private final Bitmap srcPendingUpload = mock(Bitmap.class);
    private final MarkerBitmap markerBitmap = new MarkerBitmap(mock(Context.class),
            srcWithoutPhoto, srcWithPhoto, srcOwnPhoto, srcWithoutPhotoInactive, srcWithPhotoInactive, srcOwnPhotoInactive, srcPendingUpload,
            mock(Point.class), 0, 0, mock(Paint.class));

    @Test
    public void getBitmap_OwnPhotoActive() {
        assertThat(markerBitmap.getBitmap(true, true, true, false)).isSameAs(srcOwnPhoto);
    }

    @Test
    public void getBitmap_OwnPhotoInactive() {
        assertThat(markerBitmap.getBitmap(true, true, false, false)).isSameAs(srcOwnPhotoInactive);
    }

    @Test
    public void getBitmap_PhotoActive() {
        assertThat(markerBitmap.getBitmap(true, false, true, false)).isSameAs(srcWithPhoto);
    }

    @Test
    public void getBitmap_PhotoInactive() {
        assertThat(markerBitmap.getBitmap(true, false, false, false)).isSameAs(srcWithPhotoInactive);
    }

    @Test
    public void getBitmap_WithoutPhotoActive() {
        assertThat(markerBitmap.getBitmap(false, false, true, false)).isSameAs(srcWithoutPhoto);
    }

    @Test
    public void getBitmap_WithoutPhotoInactive() {
        assertThat(markerBitmap.getBitmap(false, false, false, false)).isSameAs(srcWithoutPhotoInactive);
    }

    @Test
    public void getBitmap_PendingUpload() {
        assertThat(markerBitmap.getBitmap(false, false, false, true)).isSameAs(srcPendingUpload);
    }

}

File: app/src/test/kotlin/de/bahnhoefe/deutschlands/bahnhofsfotos/mapsforge/MarkerBitmapTest.kt
package de.bahnhoefe.deutschlands.bahnhofsfotos.mapsforge

import io.mockk.mockk
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.mapsforge.core.graphics.Bitmap

/**
 * Tests for [MarkerBitmap].
 */
class MarkerBitmapTest {
    private val srcWithoutPhoto = mockk<Bitmap>()
    private val srcWithPhoto = mockk<Bitmap>()
    private val srcOwnPhoto = mockk<Bitmap>()
    private val srcWithoutPhotoInactive = mockk<Bitmap>()
    private val srcWithPhotoInactive = mockk<Bitmap>()
    private val srcOwnPhotoInactive = mockk<Bitmap>()
    private val srcPendingUpload = mockk<Bitmap>()
    private val markerBitmap = MarkerBitmap(
        mockk(relaxed = true),
        srcWithoutPhoto,
        srcWithPhoto,
        srcOwnPhoto,
        srcWithoutPhotoInactive,
        srcWithPhotoInactive,
        srcOwnPhotoInactive,
        srcPendingUpload,
        mockk(relaxed = true),
        0f,
        0,
        mockk(relaxed = true)
    )

    @Test
    fun bitmap_OwnPhotoActive() {
        assertThat(
            markerBitmap.getBitmap(
                hasPhoto = true,
                ownPhoto = true,
                stationActive = true,
                inbox = false
            )
        )
            .isSameAs(srcOwnPhoto)
    }

    @Test
    fun bitmap_OwnPhotoInactive() {
        assertThat(
            markerBitmap.getBitmap(
                hasPhoto = true,
                ownPhoto = true,
                stationActive = false,
                inbox = false
            )
        )
            .isSameAs(srcOwnPhotoInactive)
    }

    @Test
    fun bitmap_PhotoActive() {
        assertThat(
            markerBitmap.getBitmap(
                hasPhoto = true,
                ownPhoto = false,
                stationActive = true,
                inbox = false
            )
        )
            .isSameAs(srcWithPhoto)
    }

    @Test
    fun bitmap_PhotoInactive() {
        assertThat(
            markerBitmap.getBitmap(
                hasPhoto = true,
                ownPhoto = false,
                stationActive = false,
                inbox = false
            )
        )
            .isSameAs(srcWithPhotoInactive)
    }

    @Test
    fun bitmap_WithoutPhotoActive() {
        assertThat(
            markerBitmap.getBitmap(
                hasPhoto = false,
                ownPhoto = false,
                stationActive = true,
                inbox = false
            )
        )
            .isSameAs(srcWithoutPhoto)
    }

    @Test
    fun bitmap_WithoutPhotoInactive() {
        assertThat(
            markerBitmap.getBitmap(
                hasPhoto = false,
                ownPhoto = false,
                stationActive = false,
                inbox = false
            )
        )
            .isSameAs(srcWithoutPhotoInactive)
    }

    @Test
    fun bitmap_PendingUpload() {
        assertThat(
            markerBitmap.getBitmap(
                hasPhoto = false,
                ownPhoto = false,
                stationActive = false,
                inbox = true,
            )
        )
            .isSameAs(srcPendingUpload)
    }
}